File: src/main/java/me/testaccount666/serversystem/commands/executables/lightning/TabCompleterLightning.java
package me.testaccount666.serversystem.commands.executables.lightning;

import me.testaccount666.serversystem.commands.interfaces.ServerSystemTabCompleter;
import me.testaccount666.serversystem.userdata.User;
import org.bukkit.command.Command;

import java.util.List;
import java.util.Optional;

public class TabCompleterLightning implements ServerSystemTabCompleter {

    @Override
    public Optional<List<String>> tabComplete(User commandSender, Command command, String label, String... arguments) {
        if (arguments.length == 1) {
            if (!"visual".startsWith(arguments[0].toLowerCase())) return Optional.of(List.of());

            return Optional.of(List.of("visual"));
        }

        return Optional.empty();
    }
}

File: src/main/kotlin/me/testaccount666/serversystem/commands/executables/lightning/TabCompleterLightning.kt
package me.testaccount666.serversystem.commands.executables.lightning

import me.testaccount666.serversystem.commands.interfaces.ServerSystemTabCompleter
import me.testaccount666.serversystem.userdata.User
import org.bukkit.Bukkit
import org.bukkit.command.Command

class TabCompleterLightning : ServerSystemTabCompleter {
    override fun tabComplete(commandSender: User, command: Command, label: String, vararg arguments: String): List<String> {
        if (arguments.isEmpty() || arguments.size > 2) return listOf()
        if (arguments.size == 2) return listOf("--visual", "-v").filter { it.startsWith(arguments[1], true) }

        val players = Bukkit.getOnlinePlayers().filter { commandSender.getPlayer()?.canSee(it) ?: false }.map { it.name }.toMutableList()
        players.add("-v")
        players.add("--visual")
        return players.filter { it.startsWith(arguments[0], true) }
    }
}
